### ISSUE TITLE
Translation in other languages

### ISSUE THREAD

**Issue by g3r4n** [kamranahmedse/design-patterns-for-humans#43]
I'm thinking about translate this repository in French. 
What is the best way to do it ?

**Comment by pedrotst** [comment]
And I to brazilian portuguese

**Comment by kamranahmedse** [comment]
@g3r4n @pedrotst thanks for the offer to help. 
Best way is to wait till I complete https://github.com/kamranahmedse/design-patterns-for-humans/issues/10 :-)

**Comment by alibonbon** [comment]
[Brazzers](https://fagken.com)

**Comment by jacksonfdam** [comment]
I will help to translate to brazilian portuguese

**Comment by kamranahmedse** [comment]
Thank you. I will be in touch :)

**Comment by nertus** [comment]
If you need I can add Russian translation

**Comment by gueroverdePagoFacil** [comment]
if you need y can add Spanish translation

**Comment by gueroverde** [comment]
If you need I can add Spanish translation

**Comment by kamranahmedse** [comment]
Thank you all, you guys rock :-)

I will be in touch.

**Comment by JohnTsaiAndroid** [comment]
I will help to translate to Chinese

**Comment by truonghoangnguyen** [comment]
I can translate to Vietnamese

**Comment by aloisdg** [comment]
@g3r4n Ready for a french translation :)

**Comment by crane-yuan** [comment]
@JohnTsaiAndroid 你最近有时间吗？要不我们一起合作把这个翻译成中文版的 :smile:

**Comment by VictorAvelar** [comment]
I can help @gueroverde with the translation. :mexico: 👍

**Comment by riccardomessineo** [comment]
I can help making a mandolino-pizza version of it :)

**Comment by diogomoreira** [comment]
I can help translate to brazilian portuguese :)

**Comment by nurycaroline** [comment]
I can help translate to brazilian portuguese too :D

**Comment by dieguito151** [comment]
I can help with the Spanish translation as well. Some people could translate and others could review.

**Comment by haiiiiiyun** [comment]
@crane-yuan @JohnTsaiAndroid   I didn't find a Chinese translation project around github, so I created one: https://github.com/haiiiiiyun/design-patterns-for-humans-cn. We'll merge back when ready.

The Creational Design Patterns part has already translated, and welcome to join.

**Comment by seslak** [comment]
I can help with Serbian translation.

**Comment by elityty** [comment]
I can help with Bulgarian translation.

**Comment by paulofreitas** [comment]
I can also help translating to Brazilian Portuguese. 😉

**Comment by dafian47** [comment]
I can help with Indonesian translation...

**Comment by Paradzayi** [comment]
I can help translating to Shona

**Comment by apetrovYa** [comment]
Hands on for italian and russian translations, happy to help :)

**Comment by kamranahmedse** [comment]
Guys you may start working on  the translations like this https://github.com/kamranahmedse/design-patterns-for-humans/pull/73.

Actually I had this on hold because I wanted to do it after the website. But it is fine we will move the translations once the website is done.

Thank you 👍

**Comment by gabsprates** [comment]
@pedrotst did you start the translation? I want to help too.

**Comment by sergeylunev** [comment]
@nertus hey, do you start with translation to russian?

**Comment by Egorvoz** [comment]
Happy to help with translation to Russian.

**Comment by phastmike** [comment]
Hi, my approach was not to fork and change the code. Instead, i made **running** examples of the given design patterns in Vala:
[design-patterns-for-humans in Vala](https://github.com/phastmike/vala-design-patterns-for-humans).

**Comment by mandaputra8** [comment]
Happy to help with indonesian translation

**Comment by TxGVNN** [comment]
Consider using http://weblate.org to allow everyone can translate easily.

**Comment by Egorvoz** [comment]
@TxGVNN it will be awesome dude.

**Comment by ivandrofly** [comment]
Hello guys.
Well, I have a suggestion it would be better if this repository is placed into an organization, by doing to there can be a team for each language which will make things more manageable.

**Comment by aloisdg** [comment]
@ivandrofly btw [any help is welcome](https://github.com/aloisdg/design-patterns-for-humans-cs) ;)

**Comment by ygorlazaro** [comment]
@kamranahmedse I can work with portuguese as well

**Comment by Egorvoz** [comment]
@ivandrofly your idea is great. So who can create organization ? Can you ?

**Comment by ivandrofly** [comment]
@Egorvoz done  https://github.com/design-patterns-for-humans

I may need some of you guys to help setup the organization.
@kamranahmedse are you available?

**Comment by Egorvoz** [comment]
@ivandrofly how to join this organization ?

**Comment by ivandrofly** [comment]
I sent you invitation, check your email.

For anyone else please, let me know if you want to join

**Comment by VictorAvelar** [comment]
I want to join to the org, I'm a coder but right now I am working on the Spanish translation.

Regards!

**Comment by ivandrofly** [comment]
@VictorAvelar coder on which language? (check your email)

**Comment by ivandrofly** [comment]
@VictorAvelar coder on which language? (check your email)

**Comment by UsulPro** [comment]
@ivandrofly I‘d like to join as well!
currently I am working on the Russian translation and I‘m also interested in Javascript ver

**Comment by dafian47** [comment]
@ivandrofly may i join too? i'm working on the Indonesian translation
@mandaputra8 join me to translating into Indonesian language

**Comment by ivandrofly** [comment]
@dafian47 invitation sent.

For those who have already started the translation process on a private repository please move it to https://github.com/design-patterns-for-humans if you don't have access let me know.

**Comment by UsulPro** [comment]
Thanks @ivandrofly !
Russian translation starts [here](https://github.com/design-patterns-for-humans/Russian)

BTW Guys, how do you like this [Javascript version?](https://github.com/sohamkamani/javascript-design-patterns-for-humans)

**Comment by NoobTW** [comment]
@UsulPro That's Awesome

BTW, I can help translate the Chinese Traditional version.

**Comment by diogomoreira** [comment]
Thanks for the invitation @ivandrofly.

@pedrotst @jacksonfdam @nurycaroline @paulofreitas @ygorlazaro and anyone else who want to help with the Brazilian Portuguese translation, I've started the repository [here](https://github.com/design-patterns-for-humans/brazilian-portuguese).

**Comment by ygorlazaro** [comment]
Hi Diogo,

I've already started a repo, github.com/ygorlazaro/js-stack-from-scratch

________________________________

**Comment by phastmike** [comment]
I think that there are two kinds of 'language' translations which should being done, real languages (spoken ones) and programming languages. @ivandrofly pse invite me (i'm more into programming languages translation, but can help with portuguese)

**Comment by ivandrofly** [comment]
> I think that there are two kinds of 'language' translations which are being done, real languages (spoken ones) and programming languages.

Exactly we have two types of teams

Invitation sent, I added you to pt-br-team (for natural language) 

> (i'm more into programming languages translation,

What programming language?

**Comment by phastmike** [comment]
Obrigado @ivandrofly 

> What programming language?

I did it in [Vala](https://github.com/phastmike/vala-design-patterns-for-humans) but the code segments, not the description. Vala is one of the GNOME languages with a C#/Java like syntax.

**Comment by ivandrofly** [comment]
De nada :)

Okay, start by transferring your repository to https://github.com/design-patterns-for-humans.

**Comment by XPanniX** [comment]
I'd like to help to translate all the texts to german :)

Could someone give a quick intro on how to get started?

**Comment by johannesschobel** [comment]
I can help with german too..

**Comment by vincent-paing** [comment]
I'd be more than happy to help with Burmese

**Comment by KomaBeyond** [comment]
@crane-yuan 跟我合作一起翻译中文版的吧，我开始翻译了一段时间，后面因为一些事儿一直耽搁到现在

**Comment by crane-yuan** [comment]
@KomaBeyond 不好意思，我最近有点忙，所以不能够和你一起合作翻译了，另外，已经有热心的开发者翻译成中文了，如果你有兴趣可以去联系他 @haiiiiiyun  https://github.com/haiiiiiyun/design-patterns-for-humans-cn

**Comment by KomaBeyond** [comment]
@crane-yuan 好的

**Comment by 73VW** [comment]
I can and will help with French translation too.
I can also ask my classmates (we are software engineering students) if they want to take part if you want.

**Comment by ivandrofly** [comment]
> I can and will help with French translation too.
> I can also ask my classmates (we are software engineering students) if they want to take part if you want.

Invitation sent :)

**Comment by 73VW** [comment]
@ivandrofly is there a discord or slack for the organization?

**Comment by ivandrofly** [comment]
> @ivandrofly is there a discord or slack for the organization?

https://designpatterns4humans.slack.com/

post your email address so I can add you :)... members aren't active very often but if you have any urgent  thing you want to discus with you team you can open a issue...

**Comment by jacksonfdam** [comment]
Add me please.

att.
Jackson F. de A. Mafra
iOS / Android / Front-End / Back-End Software Engineer
http://about.me/jacksonfdam | http://cvgram.me/jacksonfdam
https://www.vizify.com/jacksonfdam
twitter: @jacksonfdam | LinkedIn: http://linkedin.com/in/jacksonfdam

**Comment by 73VW** [comment]
@ivandrofly add mael.pedretti@he-arc.ch and poke me when I have been added, I will delete this comment.

**Comment by j-ew-s** [comment]
Hi, @ivandrofly  could you add me too ?

**Comment by ivandrofly** [comment]
> Hi, @ivandrofly could you add me too ?

which team (language?

**Comment by ivandrofly** [comment]
> @ivandrofly add mael.pedretti@he-arc.ch and poke me when I have been added, I will delete this comment. Thanks in advance

sent

**Comment by j-ew-s** [comment]
@ivandrofly   Portugues - Brazil

**Comment by jacksonfdam** [comment]
@ivandrofly <https://github.com/ivandrofly> Portugues - Brazil too!

att.
Jackson F. de A. Mafra
iOS / Android / Front-End / Back-End Software Engineer
http://about.me/jacksonfdam | http://cvgram.me/jacksonfdam
https://www.vizify.com/jacksonfdam
twitter: @jacksonfdam | LinkedIn: http://linkedin.com/in/jacksonfdam

**Comment by ivandrofly** [comment]
Both added /cc @j-ew-s @jacksonfdam

**Comment by j-ew-s** [comment]
my email: gabriel.scavassa@outlook.com . when done ping me . i'll delete this post.

**Comment by ivandrofly** [comment]
> my email: gabriel.scavassa@outlook.com . when done ping me @ivandrofly . i'll delete this post.

Invitation sent!

**Comment by kamranahmedse** [comment]
Hey guys, I got busy with other stuff and wasn't able to focus on this repository. I have started working on improving it again.

For the translations, this issue got a lot messy so I am going to close it. If anyone wants to work on the translations, please create a PR with `README-language.md`. 

Thanks
